engine/src/main/java/io/camunda/zeebe/engine/processing/common/CatchEventBehavior.java

```diff
@@ -89,10 +90,24 @@ public void unsubscribeFromEvents(
       final ExecutableCatchEventSupplier supplier,
       final SideEffects sideEffects,
       final TypedCommandWriter commandWriter) {
+    return subscribeToEvents(context, supplier, sideEffects, commandWriter, null);
+  }
+
+  /** @return either a failure or nothing */
+  public Either<Failure, Void> subscribeToEvents(
+      final BpmnElementContext context,
+      final ExecutableCatchEventSupplier supplier,
+      final SideEffects sideEffects,
+      final TypedCommandWriter commandWriter,
+      final VariablesLookup secondaryLookup) {
+
+    final var expressionProcessorToUse =
+        secondaryLookup == null ? expressionProcessor : expressionProcessor.chain(secondaryLookup);
```

Comment:
💭  Reading the method call here, I wonder if we can improve the understandability by making it clear what chaining the expression processor means. Perhaps something like `expressionProcessor.withChainedLookup(secondaryLookup)` or `expressionProcessor.withExtraLookup(secondaryLookup)` or something else altogether.

Reply:
Agreed:

method naming:
```
  public ExpressionProcessor withSecondaryLookup(final VariablesLookup secondaryLookup) {
    return new ExpressionProcessor(
        expressionLanguage, evaluationContext.chain(secondaryLookup));
  }

  public ExpressionProcessor withPrimaryLookup(final VariablesLookup secondaryLookup) {
    
  }
```

- chaining is done on `EvlauationContext`